java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,257 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Arguments for {@link glide.api.commands.SortedSetBaseCommands#zrange} and {@link
+ * glide.api.commands.SortedSetBaseCommands#zrangeWithScores}
+ *
+ * @see <a href="https://redis.io/commands/zrange/">redis.io</a>
+ */
+public class RangeOptions {
+
+    public interface ScoreRange {
+        String toArgs();
+    }
+
+    /** Enumeration representing numeric positive and negative infinity bounds for a sorted set. */
+    @RequiredArgsConstructor
+    public enum InfScoreBound implements ScoreRange {
+        POSITIVE_INFINITY("+inf"),
+        NEGATIVE_INFINITY("-inf");
+
+        private final String redisApi;
+
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific numeric score boundary in a sorted set. */
+    public static class ScoreBoundary implements ScoreRange {
+        /** The score value. */
+        private final double bound;
+
+        /** Whether the score value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public ScoreBoundary(double bound, boolean isInclusive) {
```

Comment:
add javadocs for constructors

Reply:
I don't think there is a need for that the instance variables have documentation plus the class declaration has documentation.

Reply:
Unfortunatenly, it is not shown in an IDE when you open a parethesis to call a constructor.